### ISSUE TITLE
Restore fails for vcxproj projects with ProjectName tag

### ISSUE THREAD

**Issue by ddrinka** [NuGet/Home#2535]
Restoring packages from project.json fails when the `<ProjectName>` tag is included in a C++/CLI vcxproj project file.  This appears to have been introduced in the newest NuGet release as it worked fine in the past.

The following is output from MSBuild:

```
Restoring packages for 'NuGetFail'.
Restoring packages for C:\Users\ddrinka\Documents\Visual Studio 2015\Projects\NuGetFailDemo\NuGetFailDemo\project.json...
Error occurred while restoring NuGet packages: System.InvalidOperationException: Missing external reference metadata for NuGetFail
   at NuGet.Commands.RestoreCommand.<ExecuteRestoreAsync>d__11.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
...
NuGet Package restore finished for project 'NuGetFail'.
NuGet package restore failed.
```

Versions:
Visual Studio 14.0.25123.00 Update 2
NuGet 3.4.1

**Comment by yishaigalatzer** [comment]
@ddrinka can you provide a small repro project?

**Comment by emgarten** [comment]
@ddrinka were references from nuget packages actually working for you when using project.json with C++?

**Comment by ddrinka** [comment]
Yes, they work great besides this bug.  The flow is much better than with
packages.config which did not support C++/CLI well.

**Comment by ddrinka** [comment]
Here's a repro:
https://github.com/johnny-robot/NugetIssue2535

**Comment by emgarten** [comment]
Thanks for the repro. This will be fixed in 3.4.3

NuGet/NuGet.Client@066c2a67eb97519ffa28ddb6df892816a5115ef3
